executor/adapter.go

```diff
@@ -282,6 +284,18 @@ func (a *ExecStmt) RebuildPlan(ctx context.Context) (int64, error) {
 	return is.SchemaMetaVersion(), nil
 }
 
+func (a *ExecStmt) setPProfLabelWithPlan(ctx context.Context) context.Context {
+	if a.Plan == nil || !config.TopSQLEnabled() {
+		return ctx
+	}
+	_, sqlDigest := a.Ctx.GetSessionVars().StmtCtx.SQLDigest()
+	normalizedPlan, planDigest := getPlanDigest(a.Ctx, a.Plan)
+	if len(planDigest) > 0 {
+		ctx = tracecpu.SetGoroutineLabelsWithSQLAndPlan(ctx, sqlDigest.String(), planDigest, normalizedPlan)
```

Comment:
Should we exclude plan text from the label?

Reply:
The plan text use to register into the collector instead of being a label.